pallets/staking/src/lib.rs

```diff
@@ -1345,8 +1369,9 @@ decl_storage! {
                     T::Currency::free_balance(&stash) >= balance,
                     "Stash does not have enough balance to bond."
                 );
+                let controller_origin = <Module<T>>::get_origin(controller.clone());
                 let _ = <Module<T>>::bond(
-                    T::Origin::from(Some(stash.clone()).into()),
+                    <Module<T>>::get_origin(stash.clone()),
                     T::Lookup::unlookup(controller.clone()),
```

Comment:
You probably don't need this `clone` now.

Reply:
Nope it is a reference and AccountId doesn't have the copy trait so have to use the clone()